### DISCUSSION TITLE
Custom Vertex shader support?

### DISCUSSION THREAD

**Discussion by BrandonPye25** [visualdesigncafe/nature-renderer#18] [Help] [answered]
Hello I see that this product supports custom shaders but wanted to see if I could get some clarification on a few things, Will we be able to implement dynamic displacement?
 We want to run a compute shader with a grass height map in the background which will update to deform the grass when any object presses down on it. So does your system allow for us to run a custom vertex shader to use that data to deform the grass vertices? We would also like to use the same system for fire propagation.

**Comment by maxplooi** [comment, marked as answer]
Yes, you can use any custom shader, including custom vertex shaders. There are no limitations on the type of shader you can use. Our own [Nature Shaders](https://assetstore.unity.com/packages/vfx/shaders/nature-shaders-2022-266863) asset also uses custom vertex shaders for interaction and wind.

**Comment by BrandonPye25** [comment]
Thanks for your prompt response. I really appreciate it. Looking forward to
using your product.
Thanks
Brandon Pye
